agent/metadata_service.go

```diff
@@ -188,10 +197,23 @@ func (mds *metadataService) getCredentials(w http.ResponseWriter, r *http.Reques
 /*
 NewMetadataService returns a properly-initialized metadataService for use.
 */
-func NewMetadataService(listener net.Listener, creds CredentialsSource) (MetadataService, error) {
+func NewMetadataService(listener net.Listener, creds credentialsSource, extraAllowedIps []*net.IPNet) (MetadataService, error) {
+
+	allowIps := []*net.IPNet{}
+	if extraAllowedIps != nil {
+		allowIps = extraAllowedIps
+	}
+
+	// Add default allowed nets to the list
+	for _, cidr := range []string{"127.0.0.1/32", "169.254.169.254/32"} {
+		_, ipNet, _ := net.ParseCIDR(cidr)
```

Comment:
For completeness' sake / future proofing in case the IP specs are moved or changed, this error return should be checked.